src/Razor/test/Microsoft.NET.Sdk.Razor.Test/GetDefaultStaticWebAssetsBasePathTest.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.Build.Framework;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Razor.Tasks
+{
+    public class GetDefaultStaticWebAssetsBasePathTest
+    {
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        [InlineData(" ")]
+        [InlineData("   ")]
+        public void ReturnsError_WhenBasePath_DoesNotContainNonWhitespaceCharacters(string basePath)
+        {
+            // Arrange
+            var expectedError = $"Base path '{basePath ?? "(null)"}' must contain non-whitespace characters.";
+
+            var errorMessages = new List<string>();
+            var buildEngine = new Mock<IBuildEngine>();
+            buildEngine.Setup(e => e.LogErrorEvent(It.IsAny<BuildErrorEventArgs>()))
+                .Callback<BuildErrorEventArgs>(args => errorMessages.Add(args.Message));
+
+            var task = new GetDefaultStaticWebAssetsBasePath {
+                BuildEngine = buildEngine.Object,
+                BasePath = basePath
+            };
+
+            // Act
+            var result = task.Execute();
+
+            // Assert
+            Assert.False(result);
+            var message = Assert.Single(errorMessages);
+            Assert.Equal(expectedError, message);
+        }
+
+        [Theory]
+        [InlineData("Identity", "identity")]
+        [InlineData("Microsoft.AspNetCore.Identity", "microsoftaspnetcoreidentity")]
```

Comment:
Can you produce a project file with slashes? Also, do we implicitly trust these paths at runtime? Could I author a package where the BasePath is `..\appsettings.json` and start serving files outside the wwwroot?

Reply:
* We only generate a safe base path for referenced projects (for which you have total control).
* I don't think you can have slashes in your PackageIds which renders this unworkable.
* When you install a package you are trusting the package author, as I could author a package that runs random logic in your application when you perform an action.
* We will be handling the base path at runtime during **development**.
* At **publish** time it will be combined with the rest of your app, but again, the moment you 
  install a nuget package you choose to trust it. You could use this feature to achieve what you 
  suggest, but you could also do that with a custom msbuild target, so this isn't anything new.